lib/smart_city/event/dataset_update/business.ex

```diff
@@ -0,0 +1,103 @@
+defmodule SmartCity.Event.DatasetUpdate.Business do
+  @moduledoc """
+  A struct representing the business data portion of a dataset update struct definition (represented by `SmartCity.Event.DatasetUpdate`)
+
+  You probably won't need to access this module directly; `SmartCity.Event.DatasetUpdate.new/1` will build this for you
```

Comment:
I appreciate your guys' attention to detail with the documentation. Everything feels really clearly explained (except maybe the macro, but I'm not complaining).